subprojects/docs/src/snippets/java/sourceSets-badConfNames/tests/badConfNames.out

```diff
@@ -0,0 +1,7 @@
+When creating configurations during sourceSet myCode setup, Gradle found that configuration myCodeCompileClasspath already exists with permitted usage(s):
+	Consumable - this configuration can be selected by another project as a dependency
+	Resolvable - this configuration can be resolved by this project to a set of files
+	Declarable - this configuration can have dependencies added to it
+Yet Gradle expected to create it with the usage(s):
+	Resolvable - this configuration can be resolved by this project to a set of files
+Gradle will mutate the usage of configuration myCodeCompileClasspath to match the expected usage. This may cause unexpected behavior. Creating configurations with reserved names has been deprecated. This will fail with an error in Gradle 9.0. Create sourceSets prior to creating or accessing the configurations associated with them. For more information, please refer to https://docs.gradle.org/0.0.0/userguide/building_java_projects.html#sec:implicit_sourceset_configurations in the Gradle documentation.
```

Comment:
Something about this wording seems a bit off to me. Especially the `Gradle will mutate` part, since by the time the user has seen this message, the configuration has _already_ been mutated. 

Furthermore I think we should try to avoid printing the definition of the role more than once. Here, `Resolvable - this configuration can be resolved by this project to a set of files` is printed twice and is redundant. Really, I think we should be able to avoid printing the role definitions altogether. I think these get in the way of the real solution to the problem which should be front and center: `Create sourceSets prior to creating or accessing the configurations associated with them`. The user should not have to parse through all this text to find out what to do. IMO most of this text belongs in linked documentation.

Reply:
> since by the time the user has seen this message, the configuration has already been mutated.

This isn't true, it prints first, and then attempts the mutation if possible.  If there is a failure mutating devs should at least see this message first.

> Really, I think we should be able to avoid printing the role definitions altogether.

I strongly disagree here.  This is exactly the kind of message I want to see.  In order:
 
1. During X
2. Gradle found Y
3. Gradle expected Z
4. Gradle will now do A because of this, which might go sideways
5. You should make change B instead, to make doing A unnecessary
6. Link to more info

I don't think we should worry about making this concise for conciseness' sake - if they don't want to see this message then they shouldn't set up Y in their builds.  If they parse/read through it carefully it should fully explain the situation and save them some searching.  I think adding a descriptions of the different roles came from Sterling during earlier configuration work.  I do wish there was more control over putting newlines in line 7.